config.tutorial.yaml

```diff
@@ -337,18 +337,39 @@ costs:
 
 
 monte_carlo:
+  # Uniform: https://chaospy.readthedocs.io/en/master/api/chaospy.Uniform.html
+  # Normal: https://chaospy.readthedocs.io/en/master/api/chaospy.Normal.html
+  # LogNormal: https://chaospy.readthedocs.io/en/master/api/chaospy.LogNormal.html
+  # Triange: https://chaospy.readthedocs.io/en/master/api/chaospy.Triangle.html
+  # Beta: https://chaospy.readthedocs.io/en/master/api/chaospy.Beta.html
+  # Gamma: https://chaospy.readthedocs.io/en/master/api/chaospy.Gamma.html
+  # [mean, std] for Normal and LogNormal
+  # [lower_bound, upper_bound] for Uniform
+  # [lower_bound, midpoint, upper_bound] for Triangle
+  # [alpha, beta] for Beta
+  # [shape, scale] for Gamma
+  # TODO: Support inputs to simulate outages biggest lines "lines.p_nom_opt.max(): [-3000MW 0MW]""
+  # TODO: Support inputs to simulate outages of biggest power plant "generators.p_nom.max()": [-1000MW 0MW]
+  # ... user can add flexibly more features for the Monte-Carlo sampling
+  # as key: add below the pypsa object for the monte_carlo sampling, "network" is only allowed for filtering!
+  # as value: currently supported format [l_bound, u_bound] or empty [], represent multiplication factors for the object
   options:
     add_to_snakefile: false
-    samples: 7    # number of optimizations
-    sampling_strategy: "chaospy"    # "pydoe2", "chaospy", "scipy", packages that are supported
-  pypsa_standard:
-      # User can add here flexibly more features for the Monte-Carlo sampling.
-      # Given as "key: value" format
-      # Key: add below the pypsa object for the monte_carlo sampling, "network" is only allowed for filtering!
-      # Value: currently supported format [l_bound, u_bound] or empty [], represent multiplication factors for the object
-    loads_t.p_set: [0.9, 1.1]
-    # generators_t.p_max_pu.loc[:, n.generators.carrier == "wind"]: [0.9, 1.1]
-    # generators_t.p_max_pu.loc[:, n.generators.carrier == "solar"]: [0.9, 1.1]
+    samples: 9 # number of optimizations. Note that number of samples when using scipy has to be the square of a prime number
+    sampling_strategy: "chaospy"  # "pydoe2", "chaospy", "scipy", packages that are supported
+  uncertainties:
+    loads_t.p_set:
+      scale: [0.9, 1.1]
+      type: Uniform
+      args: [0, 1]
```

Comment:
Nice proposal, as a comment:
to me, the use of scale and args, seem to be a duplication in the definition of uncertainties.
For example, for uniform, we could use directly:  args: [0.9, 1.1], or even explicitly state: lower_bound: 0.9, upper_bound: 1.1 (keys matching the arguments of the scipy function for example).
What do you think?

Reply:
True. Thats a clear point. Thank you very much. Args is meant to be the parameters for the chosen distribution.

Reply:
Mmm not sure if it is just a naming convention.
If we dropped entirely the scale factor and use args: [0.9, 1.1], I'd expect the uniform distribution to be created between 0.9 and 1.1.
A similar behavior may be expected for the other distributions.
The idea is that the arguments already describe the values of the distribution.

The concept of bounds is different and may apply for some continuous distributions (e.g. gaussian) when we may want to truncate some values, but the concepts are different and I'm unsure we want to delve into that.

By looking at the code, probably the scale is kind-of due to historical reasons but may be now revised.
For each scenario, we always first draw a vector of uniform distributions between 0 and 1 for every uncertain parameter, then we convert the uniform sampling into the desired distribution using the ppf methodology.
The scale factor is no more needed as the first extraction is always between 0 and 1 and then, the uniform sampling is mapped into the desired distribution using the inverse function.
The arguments (args) of each distribution already account for the shape of the desired distribution.
Does it make sense?

Reply:
I think args is particularly useful to vary the shape of the desired distribution of the user when the desired chosen distribution is either gamma, beta, lognormal, normal, or Triangular. 

The goal here is to give users the flexibility to control the shape of selected distributions that aren't uniform.

Reply:
the standard uniform yes, but in our application the lower and upper bounds may change as discussed and reviewed.
Resolving this conversation